### PR TITLE
Migrated default baseline to module config #809

### LINKED ISSUE #809: Migrate default baseline to module configuration
Currently a default baseline can be configured by editing the module manifest.

```powershell
@{
  # Private data to pass to the module specified in RootModule/ModuleToProcess. This may also contain a PSData hashtable with additional module metadata used by PowerShell.
  PrivateData = @{
      PSData = @{
          # Tags applied to this module. These help with module discovery in online galleries.
          Tags = @('PSRule-rules')
      }
      PSRule = @{
          Baseline = 'Enterprise.Default'
      }
  }
}
```

Adding a PSRule property in the manifest is problematic because the `Update-ModuleManifest` cmdlet reformats this.

Module configuration support was added later which provides an alternative mechanism to configure the default baseline.

For example:

```yaml
---
# Synopsis: Example module configuration for Enterprise.Rules module
apiVersion: github.com/microsoft/PSRule/v1
kind: ModuleConfig
metadata:
  name: Enterprise.Rules
spec:
  rule:
    baseline: 'Enterprise.Default'
```

If default baseline in both module manifest and module configuration are set, Module configuration should take priority.

### PR DESCRIPTION
**Pull request by BernieWhite**
## PR Summary

- Migrated default baseline to module configuration.
    - This enables configuration of the default baseline for a module with a module configuration.
    - This depreciate configuring the default baseline within the module manifest.
    - Modules using manifest configuration will start warning from v1.9.0.

Fixes #809

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Code changes**
  - [x] Have unit tests created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Microsoft/PSRule/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule/Rules/RuleFilter.cs [hunk 1]
```diff
@@ -95,14 +94,18 @@ private bool IsIncluded(string name, ResourceTags tag)
...
-        private bool MatchWildcard(string name)
+        private bool Contains(string name, string[] set)
         {
-            if (_WildcardMatch == null)
-            {
-                return false;
-            }
+            for (var i = 0; set != null && i < set.Length; i++)
+                if (ResourceIdEqualityComparer.IdEquals(name, set[i]))
+                    return true;
```
Comment (ArmaanMcleod) [inline src/PSRule/Rules/RuleFilter.cs:101]: I wonder if we can use LINQ here: `set.Any(item => ResourceIdEqualityComparer.IdEquals(name, item))` here? Unless we are avoiding LINQ for performance impact.

File: tests/PSRule.Tests/PSRule.Baseline.Tests.ps1 [hunk 2]
```diff
@@ -29,7 +29,18 @@ BeforeAll {
 
     #region TestCases
 
-    $allYamlBaselines = @"
+    $allYamlBaselines = Get-Content -Path (Join-Path -Path $here -ChildPath 'Output.Baseline.yaml') -Raw;
+    $allFourSpaceJsonBaslines = Get-Content -Path (Join-Path -Path $here -ChildPath 'Output.Baseline.4.jsonc') -Raw;
+    $allTwoSpaceJsonBaslines = Get-Content -Path (Join-Path -Path $here -ChildPath 'Output.Baseline.2.jsonc') -Raw;
+    $allZeroSpaceJsonBaslines = Get-Content -Path (Join-Path -Path $here -ChildPath 'Output.Baseline.0.jsonc') -Raw;
```
Comment (ArmaanMcleod) [inline tests/PSRule.Tests/PSRule.Baseline.Tests.ps1:35]: This is great 🚀 . I wanted to do this before and never got around to it for some reason 😄.

**Review by ArmaanMcleod — COMMENTED** [review]
Looks great @BernieWhite. Nothing much from me, just some general comments.

**Review by ArmaanMcleod — APPROVED** [review]

File: src/PSRule/Rules/RuleFilter.cs (hunk 1 above)
Comment (BernieWhite) [reply src/PSRule/Rules/RuleFilter.cs:101]: Yep. Faster and less memory is better for low complexity solutions. But good suggestion.

**Pull request merged**